cdap-docs/developers-manual/source/building-blocks/schedules.rst

```diff
@@ -23,8 +34,96 @@ Optionally, you can specify the properties for the Schedule::
     scheduleProperties.put("myProperty", "10");
     scheduleProperties.put("anotherProperty", "anotherValue");
 
-    scheduleWorkflow("FiveHourSchedule", "0 */5 * * *", "MyWorkflow", scheduleProperties);
+    scheduleWorkflow(Schedules.createTimeSchedule("FiveHourSchedule", "Schedule running every 5 hours", "0 */5 * * *"),
+                     "MyWorkflow", scheduleProperties);
     ...
 
 Every time the ``FiveHourSchedule`` triggers, it passes the ``scheduleProperties`` as runtime arguments to the ``MyWorkflow``.
 The properties specified by the Schedule can be overridden by the Workflow's runtime arguments, with the Workflow's having precedence.
+
+
+.. _stream-size-schedules:
+
+Stream Size Schedules
+=====================
+
+.. rubric:: Definition and Usage of Stream Size Schedules
+
+**Stream Size Schedules** will execute based on data ingested in :ref:`Streams <streams>`, using the
+`createDataSchedule API <../../reference-manual/javadocs/co/cask/cdap/api/schedule/Schedules.html#createDataSchedule(java.lang.String,%20java.lang.String,%20co.cask.cdap.api.schedule.Source,%20java.lang.String,%20int)>`__.
+Here is an example to add a **Stream Size Schedule** based on a Stream named ``purchaseStream`` that triggers
+every time the Stream ingests 1MB of data::
+
+    scheduleWorkflow(Schedules.createDataSchedule("1MBStreamSchedule", "Schedule triggered every 1MB of ingested data",
+                                                  Schedules.Source.STREAM, "purchaseStream", 1),
+                     "MyWorkflow");
+
+The ``purchaseStream`` will either have to exist in CDAP when deploying your application, or you will have to
+`add it <../../reference-manual/javadocs/co/cask/cdap/api/app/AbstractApplication.html#addStream(co.cask.cdap.api.data.stream.Stream)>`__
+when configuring your application. If you try to create a **Stream Size Schedule** on a Stream which does not exist,
+your application will fail to deploy.
+
+Optionally, you can specify the properties for the Schedule::
```

Comment:
Bad copy/past from existing doc, I will reformulate to: "you can optionally specify..."